scripts_src/headers/exppoint.h

```diff
@@ -488,6 +488,6 @@
  points he received.
 ***************************************/
 #define give_xp(x)                              give_exp_points(x);             \
-                                                display_msg(message_str(SCRIPT_GENERIC,100)+x+message_str(SCRIPT_GENERIC,101))
+                                                display_msg(g_mstr(100) + (x + ((x * 5 / 100) * dude_perk(PERK_swift_learner))) + g_mstr(101))
```

Comment:
I suggest splitting visible exp calculation to separate macro for use with non-standard messages.

Reply:
Like this?
```cpp
#define calc_display_xp(x)    (x + ((x * 5 / 100) * dude_perk(PERK_swift_learner)))
#define give_xp(x)            give_exp_points(x); \
                              display_msg(g_mstr(100) + calc_display_xp(x) + g_mstr(101))
```

Reply:
Yes, or maybe `calc_xp_gain_after_perks` or `get_actual_xp_gain`. Or maybe we should really add some new opcode for this ))